### PR TITLE
fix: negotiator error message and switch to new negotiator API endpoint

### PR DESCRIPTION
**Pull request by konstantina-gkp**
fixes: #3893; #4045

What are the main changes you did:
- Display an error message in the checkout when negotiator is unavailable.

how to test:
- Go to the directory app.
- Add a collection to the negotiator selection.
- Press "Request" and see error message.

### CODE REVIEW

**Review by dtroelofsprins — CHANGES_REQUESTED** [review]
Tested:
- fake negotiator URL: message displayed = Load failed
- Current URL in directory-emx2-acc: Doesn't work in preview, message is: Negotiator is not available. Please try again later.
- The default one in the app ("https://negotiator.acc.bbmri-eric.eu/api/v3/requests",) => Negotiator is not available. Please try again later.

The last two could make sense as the preview url is not known in the negotiator service. 

However, besides the message displayed in case of failing negotiator you also mention that it partly solves 4045:
1.) Can I still use the old link with the code adjustments in the PR? 
2.) And/or does the new negotiator endpoint works with the adjustments made in this PR. But why is then 4045 only partly solved?

And one last thing regarding the displayed message, I was a bit doubting about the "Please try again later", but I guess that from a users point of view this is OK.
Then 'support' needs to know that there is probably something wrong with the negotiator URL and that just waiting will not solve it, but URL needs to be adjusted (or server needs to be 'assigned' to the negotiator).

File: apps/directory/src/components/popovers/NegotiatorSelection.vue [hunk 1]
```diff
@@ -139,9 +147,15 @@ export default {
...
-      this.cartVisible = false;
-      this.checkoutStore.sendToNegotiator();
+    async sendRequest() {
+      this.errorMessage = ""; // Reset error message before request
+      try {
+        await this.checkoutStore.sendToNegotiator();
+        this.cartVisible = false;
+      } catch (err) {
+        this.errorMessage =
+          err.message || "An error occurred. Please try again later.";
```
Comment (esthervanenckevort) [inline apps/directory/src/components/popovers/NegotiatorSelection.vue:157]: Exposing a backend error is not useful for the end user and can break the UI. So always use the generic error message. Logging the actual error to the console is okay.

**Review by esthervanenckevort — CHANGES_REQUESTED** [review]
Small change required and we need to be able to test the happy flow as well before merging.

**Comment by dtroelofsprins** [comment]
It seems now always the "An error occurred. Please try again later." message is thrown. I think @esthervanenckevort requested to always throw the generic message: "Negotiator is not available. Please try again later."

**Review by dtroelofsprins — APPROVED** [review]

**Pull request merged**